lib/grid.rb

```diff
@@ -17,9 +33,31 @@ def get_value(coordinate)
     structure[coordinate]
   end
 
+  def get_valid_options(computer_choices)
+    structure.each { |key, value|
+    if value == " "
+      computer_choices << key
+    end
+    }
+  end
+
   private
 
-  def grid_structure
-    {"A1"=> ' ', "A2"=> ' ', "A3"=> ' ', "B1"=> ' ', "B2"=> ' ', "B3"=> ' ', "C1"=> ' ', "C2"=> ' ', "C3"=> ' '}
+  def build_grid
+    coordinates = []
+    row.each do |letter|
+      column.each do |number|
+         coordinates.push(letter + number)
+      end
+    end
+    Hash[coordinates.collect { |coordinate| [coordinate, ' '] } ]
```

Comment:
I'm wondering how this programmatic build up of the grid is better than the literal? It would be useful if we were going to dynamically change the size of the grid, but that isn't this feature.

I know we had originally talked about changing the structure of the grid, but the goal was to shove data into the `Cell` object. It would have data: `coordinate` and `mark`. The coordinates would be the string row and column; the mark would be X, O or nil.